src/Domain/CoordinateMaps/Wedge3D.cpp

```diff
@@ -48,21 +50,35 @@ Wedge3D::Wedge3D(const double radius_inner, const double radius_outer,
          "The arguments passed into the constructor for Wedge3D result in an "
          "object where the "
          "outer surface is pierced by the inner surface.");
-  scaled_frustum_zero_ =
-      0.5 / sqrt(3.0) * ((1.0 - sphericity_outer_) * radius_outer +
-                         (1.0 - sphericity_inner) * radius_inner);
+  ASSERT(not with_logarithmic_map_ or
+             (with_logarithmic_map_ and sphericity_inner_ == 1.0 and
+              sphericity_outer_ == 1.0),
+         "The logarithmic map is only supported for spherical wedges.");
+  scaled_frustum_zero_ = 0.5 / sqrt(3.0) *
+                         ((1.0 - sphericity_outer_) * radius_outer +
+                          (1.0 - sphericity_inner) * radius_inner);
   sphere_zero_ = 0.5 * (sphericity_outer_ * radius_outer +
                         sphericity_inner * radius_inner);
   scaled_frustum_rate_ =
       0.5 / sqrt(3.0) * ((1.0 - sphericity_outer_) * radius_outer -
                          (1.0 - sphericity_inner) * radius_inner);
   sphere_rate_ = 0.5 * (sphericity_outer_ * radius_outer -
                         sphericity_inner * radius_inner);
+  if (with_logarithmic_map_) {
```

Comment:
In the case that `with_logarithmic_map_ == true` this undoes a bunch of work that we just did. It seems like an `if-else` style would be better here, no?